### DISCUSSION TITLE
Is it possible to add a whole group of userscripts at once from a GitHub repository?

### DISCUSSION THREAD

**Discussion by oeloo** [violentmonkey/violentmonkey#2045] [Q&A]
Hi, I am currently using the Add new URL feature of Violentmonkey to sync userscripts from my GitHub repository to Violentmonkey chrome addon.
I had to add each url one by one. Is there a way to have Violentmonkey automatically adding several userscripts at once (for example by adding a url with the list of userscripts to add) and update it when this list is changing. Regards

**Comment by tophf** [comment]
You can do it by running the following code in devtools of any Violentmonkey UI page:

```js
((urls = [
  'https://..../foo.user.js',
  'https://..../bar.user.js',
]) => Promise.all(urls.map(async url => browser.runtime.sendMessage({
  cmd: 'ParseScript',
  data: {
    code: await (await fetch(url)).text(),
    url,
  },
}))))();
```

**Reply by oeloo** [reply]
What is the"devtools of any Violentmonkey UI page"? how can I access it?

**Reply by tophf** [reply]
Click the icon for Violentmonkey in the toolbar, click the gear icon at the top to open the dashboard, open devtools via Ctrl-Shift-I or by right-clicking the page and choosing `Inspect`, then switch to the Console tab inside.

**Reply by oeloo** [reply]
In this screen where can I enter the code ? I don't see any input field
![image](https://github.com/violentmonkey/violentmonkey/assets/98819304/38939b6b-01f3-49ee-b278-62de2837a8f8)

**Reply by tophf** [reply]
It's after `>`. The cursor should be already blinking there when you open devtools console.

**Reply by oeloo** [reply]
Is it possible to store this code with the list of my userscripts inside and store in a file on github and then call it from here to add these userscripts automatically?

**Reply by tophf** [reply]
You can store it in [devtools snippets](https://developer.chrome.com/docs/devtools/javascript/snippets).

**Reply by oeloo** [reply]
Interested, I will look at this possibility. Thank you
